src/render/value.rs

```diff
@@ -1,20 +1,16 @@
 use crate::types::ast;
-use crate::types::span::Span;
 use crate::value::ValueCow;
 use crate::{Error, Result, Value};
 
 impl ValueCow<'_> {
-    pub fn as_bool(&self, source: &str, span: Span) -> Result<bool> {
+    pub fn as_bool(&self) -> bool {
         match &**self {
-            Value::Bool(cond) => Ok(*cond),
-            value => {
-                let v = value.human();
-                Err(Error::render(
-                    format!("expected bool, but expression evaluated to {v}"),
-                    source,
-                    span,
-                ))
-            }
+            Value::None | Value::Bool(false) | Value::Integer(0) => false,
+            Value::Float(n) if *n == 0.0 => false,
```

Comment:
One thought I have here is that maybe we should just check that the absolute difference is smaller than a certain value. If people are really doing floating point arithmetic and expecting it to be equal to zero that seems unlikely to ever happen.

Maybe it's worth checking what other template engines are using first?

E.g. 😕 

```rust
println!("{}", 0.000000000000000000000000000000000000000000000001 == 0.0);
# false
```

Reply:
I don't think that's a good idea; in my opinion, the truthiness function for floats is mostly there to "fill a gap" (although it could be useful in some limited sets of cases), because it would be weird if ints worked but floats didn't. JS for instance also uses a != 0.0 check for float truthiness, so you see much more often an epsilon comparison check like the one you mentioned.

The thing about baking in an epsilon to the default is that different usages can require vastly different epsilons. If you're dealing with very large numbers, for example, the epsilon in your example would be useless anyway. Conversely, if we're dealing with floats in the 0-1 range, even such absurdly small quantities may be a difference we care about. At least 0.0 is a default that has very predictable behavior (given that the user is familiar with floating point arithmetic); an arbitrary epsilon is much more confusing in edge cases, even though it may generally have the more "reasonable" behavior.

If we support user-supplied truthiness functions, they can always change this as they see fit. I think the default should still be an equality check (not a comparison check).